README.md

```diff
@@ -224,15 +225,40 @@ or `asyncQuery()`. These options control how the query runs in Fauna. See [Query
 options](#query-options).
 
 
-### Define a result class
+### Define a custom class for your data
+These should be simple POJOs.
 
-You can use the `com.fauna.annotation`  package to define a result class for a
-Fauna document. The package provides annotations like `@FaunaField` and
-`@FaunaIgnore` to map Fauna documents to Java classes and fields.
+```java
+import com.fauna.annotation.FaunaField;
+import com.fauna.annotation.FaunaId;
+
+class Person {
+    
+    @FaunaId
+    private String id;
+    
+    private String firstName;
 
-Use the `com.fauna.serialization` package to handle deserialization for
-generics, such as `PageOf`, `ListOf`, and `MapOf`.
+    @FaunaField( name = "dob")
+    private String dateOfBirth;
+}
+```
 
+You can use the `com.fauna.annotation` package to modify encoding and decoding of 
+specific fields in classes used as arguments and results of queries.
+* `@FaunaId`: Should only be used once per class and be associated with a field named `id` if it represents the ID of a document. It will not be encoded unless the `isClientGenerated` flag is set.
+* `@FaunaTs`: Should only be used once per class and be associated with a field named `ts` if it represents the timestamp of a document. It will never be encoded.
+* `@FaunaColl`: Typically goes unmodeled. Should only be used once per class and be associated with a field named `coll` if represents the collection field of a document. It will never be encoded.
+* `@FaunaField`: Can be associated with any field to override its name in Fauna.
+* `@FaunaIgnore`: Can be used to ignore fields during encoding and decoding.
+
+In the `com.fauna.codec` package, you use classes to handle type erasure when the top-level result
+of a query is a generic, including:
+* `PageOf<T>` where `T` is the element type.
+* `ListOf<T>` where `T` is the element type.
+* `MapOf<T>` where `T` is the value type.
+* `OptionalOf<T>` where `T` is the value type.
+* `NullableOf<T>` where `T` is the value type. This is specifically for cases when returning a Fauna Document that may be null and you want to receive a concrete NullDoc<T> or NonNullDoc<T> instead of catching a NullDocumentException.
```

Comment:
```suggestion
* `NullableOf<T>` where `T` is the value type. This is specifically for cases when you return a Fauna document that may be null and want to receive a concrete `NullDoc<T>` or `NonNullDoc<T>` instead of catching a `NullDocumentException`.
```